android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiWebViewClient.java

```diff
@@ -187,4 +191,39 @@ public void onReceivedSslError(WebView view, SslErrorHandler handler, SslError e
 			Log.e(TAG, "SSL error occurred: " + error.toString());
 		}
 	}
+
+	@Override
+	public void onLoadResource(WebView view, String url)
+	{
+		super.onLoadResource(view, url);
+		KrollDict data = new KrollDict();
+		data.put(TiC.PROPERTY_URL, url);
+		webView.getProxy().fireSyncEvent(TiC.EVENT_WEBVIEW_ON_LOAD_RESOURCE, data);
+	}
+
+	// This method was deprecated from Android API level 21.
+	@TargetApi(Build.VERSION_CODES.HONEYCOMB)
+	@Override
+	public WebResourceResponse shouldInterceptRequest(WebView view, String url)
+	{
+		if(Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
```

Comment:
There's actually 2 versions of 'shouldInterceptRequest'. 

'WebResourceResponse shouldInterceptRequest(WebView view, String url)' is deprecated from Lollipop.
'WebResourceResponse shouldInterceptRequest(WebView view, WebResourceRequest request)' is the new API available from Lollipop onwards to replace the deprecated one.

The 2 events fire on slightly different occasions but, at the end, for the user, it might be the same.
'shouldInterceptRequest' -> Notify the host application of a resource request
'onLoadResource' -> Notify the host application that the WebView will load the resource
Ref: http://developer.android.com/reference/android/webkit/WebViewClient.html

Thus, for this use case, would it be better to remove 'shouldInterceptRequest' and only use 'onLoadResource'? If so, I'll remove 'shouldInterceptRequest'.